backend/battle/tests/test_view.py

```diff
@@ -0,0 +1,134 @@
+from model_bakery import baker
+from common.utils.tests import TestCaseUtils
+from django.urls import reverse
+from battle.models import PokemonTeam, Team
+
+
+class TeamViewTest(TestCaseUtils):
+    def setUp(self):
+        super().setUp()
+        self.battle = baker.make('battle.Battle', creator=self.user)
+        names = ['pikachu', 'bulbasaur', 'pidgeot', 'wrong_name']
+        for pokemon_name in names:
+            baker.make("pokemon.Pokemon", name=pokemon_name, attack=60, defense=15, hp=10)
+
+    def test_user_can_create_team_with_pokemons(self):
+        pokemons_data = {
+            "battle": self.battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        response = self.auth_client.post(
+            reverse("team_create", kwargs={'pk': self.battle.id}), pokemons_data, follow=True)
+        self.assertEqual(response.status_code, 200)
+
+        team_user = Team.objects.filter(battle=self.battle, trainer=self.user)
+        self.assertTrue(team_user)
+
+        pokemons_on_team_user = []
+
+        pokemons_data_submited = [
+            pokemons_data['pokemon_1'],
+            pokemons_data['pokemon_2'],
+            pokemons_data['pokemon_3']
+        ]
+
+        all_pokemons_on_team = PokemonTeam.objects.filter(
+            team=team_user[0]).prefetch_related('pokemon')
+
+        for count in enumerate(pokemons_data_submited):
+            pokemons_on_team_user.append(all_pokemons_on_team[count[0]].pokemon.name)
+
+        self.assertEqual(pokemons_on_team_user, pokemons_data_submited)
```

Comment:
You can do here:

```
pokemons_data_submited = [
    pokemons_data['pokemon_1'],
    pokemons_data['pokemon_2'],
    pokemons_data['pokemon_3']
]
self.assertEqual(
    set(pokemons_data_submited), 
    set([p.name for p in team.pokemons.all()])
)
```

Reply:
Thanks! This will make easy many things!